server/src/test/java/org/elasticsearch/rest/action/document/RestIndexActionTests.java

```diff
@@ -84,4 +91,34 @@ private void checkAutoIdOpType(Version minClusterVersion, DocWriteRequest.OpType
         dispatchRequest(autoIdRequest);
         assertThat(executeCalled.get(), equalTo(true));
     }
+
+    public void testTypeInPath() {
```

Comment:
same comment here about failing on version 9

Reply:
this one is actually tricky.
This test is testing both compatible and current version behaviour. The RestIndexAction will be registered using its all routes, including the compatible routes. After the version bump, when the V_7 is obsolete the test will fail in `setUpAction` because of the assert in the RestController:L179 
```
        assert RestApiVersion.minimumSupported() == version || RestApiVersion.current() == version
            : "REST API compatibility is only supported for version " + RestApiVersion.minimumSupported().major;
```
I guess I will move the compat tests into separate file and use the assert there?

Reply:
the only thing that comes to mind is to have a strategy that we only bump the minimumSupported/current versions when all of the V_7 code is removed..
But that is also against whet we already assumed and had assertions in place 
Version:L124
```
  assert RestApiVersion.current().major == CURRENT.major : "RestApiVersion must be upgraded " +
            "to reflect major from Version.CURRENT [" + CURRENT.major + "]" +
            " but is still set to [" + RestApiVersion.current().major + "]";
```

Reply:
all in all this feels bigger then just a small fix and deserves a separate follow up PR - there is plenty of time before 8 actually being released

Reply:
I'm sure I completely follow, but i am fine with follow ups later on to help enusre the bump to v9 is easier. 

However, I think adding in an [assumeTrue](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html) *per Test* that shouldn't run in v9 is pretty easy to implement now..but if I am wrong +1 for followup.

Reply:
as per discussion on the sync, I gathered the conclusions on the issue https://github.com/elastic/elasticsearch/issues/70617